agent.go

```diff
@@ -699,7 +713,7 @@ func (ep *endpoint) deleteServiceInfoFromCluster(sb *sandbox, method string) err
 			if n.ingress {
 				ingressPorts = ep.ingressPorts
 			}
-			if err := c.rmServiceBinding(ep.svcName, ep.svcID, n.ID(), ep.ID(), name, ep.virtualIP, ingressPorts, ep.svcAliases, ep.myAliases, ep.Iface().Address().IP, "deleteServiceInfoFromCluster", true); err != nil {
```

Comment:
Not specific to this PR but had a concern since we are touching this area overall: Will it be safer to remove the service binding first and then call `disableServiceInNetworkDB`/`agent.networkDB.DeleteEntry`

Reply:
That's a fair question.  Since the usual order is "lookup name, issue connection request, get redirected by NAT, get load balanced, connect to server", it would make sense to disable the name lookup first and then the load balancing.  Or from a code standpoint, since we do (1) set up load balancing and then (2) set up name resolution, it makes sense to tear them down in the opposite order.   Patch was already merged, but we could create another PR to address this.    I'll confess that I haven't looked in detail at why it removal matched instead of reversed the order of "undo"s.  Would need to verify there are no issues there.

Reply:
On a bit further reflection, though, this doesn't get hit as much as I first thought.  Containers don't generally connect to the service backends through their backend service names.   They connect through the front-end load balancer.  The DNS resolution for this doesn't go away until the last backend goes away.  So at that point the connection is *supposed* to fail.   The only difference will be whether the client receives a "host unreachable" or a "name resolution failure".   And even then, only if it happens to occur within the fractions of a second between removing the front-end IP address and the DNS entry.

Reply:
Agree with the fact that any negative effect of this will be extremely short lived. The comment was more in the context of proper ordering of cleanup ops. We can address this in a potential cleanup patch later.